core/types/transaction_signing.go

```diff
@@ -103,6 +114,44 @@ type Signer interface {
 	Equal(Signer) bool
 }
 
+type EIP86Signer struct {
+	EIP155Signer
+}
+
+func NewEIP86Signer(chainId *big.Int) EIP86Signer {
+	return EIP86Signer{
+		EIP155Signer: NewEIP155Signer(chainId),
+	}
+}
+
+func (s EIP86Signer) Equal(s2 Signer) bool {
+	_, ok := s2.(EIP86Signer)
+	return ok && s.EIP155Signer.Equal(s2)
+}
+
+func isAbstractSigner(tx *Transaction) bool {
+	v, r, s := tx.RawSignatureValues()
+	return v.BitLen() == 0 && r.BitLen() == 0 && s.BitLen() == 0
```

Comment:
The EIP has changed and `v` here [needs to be the chain ID](https://github.com/ethereum/EIPs/pull/208/files#diff-95b306e6e6d2d6dbb0c5fd3ac4ea45c4R24).